src/main/java/com/vesta/service/impl/SubjectServiceImpl.java

```diff
@@ -46,4 +50,38 @@ public void delete(Long id) {
 
         repository.deleteById(id);
     }
+
+    @Override
+    public List<SubjectDto> getAllByFloorId(Long floorId) {
+        log.info("method --- getAllByFloorId");
+
+        return repository.findAllByFloorId(floorId)
+                .stream()
+                .map(converter::convert)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void addSubject(SubjectDto dto) {
```

Comment:
Rename the method "addSubject" to "create", beacause when you read the name you think that you want to add subject to somethiong which exist not to create it.